### PR TITLE
feat: Add LogicalSectorSize support for UltraSSD

### LINKED ISSUE #559: support LogicalSectorSize for UltraSSD
**Is your feature request related to a problem?/Why is this needed**

**Describe the solution you'd like in detail**

LogicalSectorSize should be with 2020-06-30/compute, when it's ready, should be supported by this driver.

We do have a feature in GA behind an AFEC flag where we expose a new disk property to customers where we let them specify whether the disk is 512 sector size enabled.
 
This is from Az cli for example:
--logical-sector-size          : Logical sector size in bytes for Ultra disks. Supported values are 512 ad 4096. 4096 is the default.

it's here in [2020-06-30/compute](https://github.com/Azure/azure-sdk-for-go/blob/928b9cbac68670876b583465f789bd3ba655777c/services/compute/mgmt/2020-06-30/compute/models.go#L1056-L1057)
```
	// LogicalSectorSize - Logical sector size in bytes for Ultra disks. Supported values are 512 ad 4096. 4096 is the default.
	LogicalSectorSize *int32 `json:"logicalSectorSize,omitempty"`
```

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by nearora-msft**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #559 

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by andyzhangx** [comment]
there is sth wrong with the regex expression, e2e test is not triggered now

**Comment by andyzhangx** [comment]
> there is sth wrong with the regex expression, e2e test is not triggered now

oh, my fault, it's there, works well

**Comment by andyzhangx** [comment]
also add a new parameter here, that will cover e2e test easily:
https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/5b81592a60c7410f9a830a36dbdb5ea50fc64d44/test/e2e/dynamic_provisioning_test.go#L130-L134

**Comment by andyzhangx** [comment]
this test failure `pull-azuredisk-csi-driver-verify` is expected since there is vendor code change, once UltraSSD e2e test pass, we could merge cloud provider change first and then do vendoring.

**Comment by nearora-msft** [comment]
> also add a new parameter here, that will cover e2e test easily:
> https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/5b81592a60c7410f9a830a36dbdb5ea50fc64d44/test/e2e/dynamic_provisioning_test.go#L130-L134

Done

File: pkg/azuredisk/controllerserver.go [hunk 1]
```diff
@@ -141,6 +142,11 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 			diskIopsReadWrite = v
 		case diskMBPSReadWriteField:
 			diskMbpsReadWrite = v
+		case logicalSectorSizeField:
+			logicalSectorSize, err = strconv.Atoi(v)
+			if err != nil {
```
Comment (andyzhangx) [inline pkg/azuredisk/controllerserver.go:147]: also add a check `logicalSectorSize > 0` here

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve
/retest
pls also add the parameter here: https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/docs/driver-parameters.md#dynamic-provisioning

**Pull request merged**